### DISCUSSION TITLE
I would like to see a watch(-w) option on biome check.

### DISCUSSION THREAD

**Discussion by araera111** [biomejs/biome#135] [Ideas]
It would be very useful if it were available.

**Comment by ematipico** [comment]
Any specifics? What should the watch do in this context? 

I know it might seem a dumb question, but it would be great if you can give us some input, so we know what you're looking for.

**Comment by araera111** [comment]
I apologize for not informing you of the details.

biome check . /src -w or --watch will monitor the file for changes and run biome check again.

Why I need watch
I lint a lot of code with biome and fix errors little by little.
I find it tedious to run check again every time I fix one.
My way is to use the -w option to fix the last error displayed and then continue linting.

**Comment by zach-betz-hln** [comment]
Hi @araera111 if you're still interested in this, you can try a DIY [solution](https://github.com/biomejs/biome/discussions/5475) by wrapping the biome CLI in a Node.js process in watch mode.

**Comment by nikelborm** [comment]
I have the same CLI workflow. I run a biome check, select a mistake to fix, jump to the file, edit it, rerun a biome check, select a new mistake to fix, go into the file, and fix. Constantly rerunning the biome for that is tedious. Would be nice to have the same [`--watch` flag as in `tsc`](https://www.typescriptlang.org/docs/handbook/configuring-watch.html), for example.

**Reply by siketyan** [reply]
Out of curiosity, why not using the LSP server on your editor/IDE?

**Reply by nikelborm** [reply]
I currently use VSCode. Let's say I have 500 code files. 20 of them have 1 or 2 warnings. I'm not aware of how to see the whole codebase's problems overview in VSCode, and I assume there's none. So I can either run the checks in a CLI or open each of the files individually to see if there are any warnings. The IDE has a "Problems overview" tab; however, there I could only find problems related to **currently open** tabs. When the file tab is closed, its problems disappear from the overview. Opening 500 tabs to get an overview of the problems is crazy. Running the CLI, clicking on about 20 filenames to open 20 tabs and start working in this mess, closing tabs one by one, maybe not super crazy, while there are only 20, but yet still uncomfortable to me. Having a living dynamic to-do list automatically rendered in the CLI, from which problems automatically disappear one by one, while I'm editing files, and solving bite-sized problems, is the best. The important part is to have each new file report in the CLI overwrite the previous report.

**Reply by nikelborm** [reply]
A particular use-case example for this is when you don't know which new files have a new problem, such as when you enable a new linting rule that was previously disabled. Warnings may appear in absolutely random places in the codebase. Or, conversely, when you disable the rule, Biome (rightfully so) starts complaining about now useless comments that were previously disabling the rule in many places, inline.

**Reply by siketyan** [reply]
Makes sense, thank you for the thorough explanation.

> I'm not aware of how to see the whole codebase's problems overview in VSCode, and I assume there's none.

Actually there is the Workspace Diagnostics request defined in the LSP specification, that can be used to pull all diagnostics within the workspace. Biome haven't implemented this yet, but I think this will resolve your use cases.

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_diagnostic
